File: languages/csharp_unity_json_properties/Object.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Object : Element
{
    [JsonProperty("aesthetics"), TextAttribute("")]
    public string aesthetics;
    [JsonProperty("weight"), Integer(0)]
    public int weight;
    [JsonProperty("amount"), Integer(0)]
    public int amount;
    [JsonProperty("parent_object"), ReferenceAttribute(typeof(Object))]
    public string parentObject;
    [JsonProperty("materials"), ReferenceAttribute(typeof(Construct), true)]
    public string materials;
    [JsonProperty("technology"), ReferenceAttribute(typeof(Construct), true)]
    public string technology;
    [JsonProperty("utility"), TextAttribute("")]
    public string utility;
    [JsonProperty("effects"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string effects;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("consumes"), ReferenceAttribute(typeof(Construct), true)]
    public string consumes;
    [JsonProperty("origins"), TextAttribute("")]
    public string origins;
    [JsonProperty("location"), ReferenceAttribute(typeof(Location))]
    public string location;
    [JsonProperty("language"), ReferenceAttribute(typeof(Language))]
    public string language;
    [JsonProperty("affinities"), ReferenceAttribute(typeof(Trait), true)]
    public string affinities;
}

File: languages/django_models/object.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Object(AbstractElementModel):

    # Form
    aesthetics = models.TextField(blank=True, null=True)
    weight = models.PositiveIntegerField(blank=True, null=True)
    amount = models.PositiveIntegerField(blank=True, null=True)
    parent_object = models.ForeignKey("Object", on_delete=models.SET_NULL, blank=True, null=True, related_name="object_parent_object")
    materials = models.ManyToManyField("Construct", blank=True, related_name="object_materials")
    technology = models.ManyToManyField("Construct", blank=True, related_name="object_technology")

    # Function
    utility = models.TextField(blank=True, null=True)
    effects = models.ManyToManyField("Phenomenon", blank=True, related_name="object_effects")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="object_abilities")
    consumes = models.ManyToManyField("Construct", blank=True, related_name="object_consumes")

    # World
    origins = models.TextField(blank=True, null=True)
    location = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="object_location")
    language = models.ForeignKey("Language", on_delete=models.SET_NULL, blank=True, null=True, related_name="object_language")
    affinities = models.ManyToManyField("Trait", blank=True, related_name="object_affinities")

    def __str__(self):
        return self.name
